src/limit_order.cpp

```diff
@@ -0,0 +1,135 @@
+#pragma once
+#include "..\include\TradingEngine\limit_order.h"
+
+#include <memory>
+using std::shared_ptr;
+
+#include <unordered_set>
+using std::unordered_set;
+
+#include "..\include\TradingEngine\transactions.h" 
+#include "..\include\TradingEngine\order_type.h"
+#include "..\include\TradingEngine\order_status.h"
+#include "..\include\TradingEngine\order.h"
+
+const shared_ptr<Order>& LimitOrder::getHead() {
+	return limit_order_head;
+}
+
+void LimitOrder::setHead(const std::shared_ptr<Order>& incoming_order) {
+	limit_order_head = incoming_order;
+}
+
+const shared_ptr<Order>& LimitOrder::getTail() {
+	return limit_order_tail;
+}
+
+void LimitOrder::setTail(const std::shared_ptr<Order>& incoming_order) {
+	limit_order_tail = incoming_order;
+}
+
+int LimitOrder::getListLength() const {
+	return limit_order_length;
+}
+
+int LimitOrder::updateListLength(const int& update_increment) {
+	return limit_order_length += update_increment;
+}
+
+void LimitOrder::addOrder(Order& incoming_order) {
+
+	/* Create pointer to incoming order */
+	const std::shared_ptr<Order>& incoming_order_ptr = std::make_shared<Order>(incoming_order);
+
+	/* Increment LinkedList length. */
+	updateListLength(1);
+
+	/* First Order */
+	if (getListLength() == 1) {
+		setHead(incoming_order_ptr);
+		setTail(incoming_order_ptr);
+	}
+	/* Otherwise, we add an order to the end of the list */
+	else {
+		incoming_order_ptr->prev_order = getTail();
+		getTail()->next_order = incoming_order_ptr;
+		setTail(incoming_order_ptr);
+	}
+}
+
+void LimitOrder::deleteOrder(const std::shared_ptr<Order>& order_to_delete) {
+
+	/* Decrement LinkedList length. */
+	updateListLength(-1);
```

Comment:
decreate after deleting not before, because at this point, you will have wrong data and if an exception happened afterwards, you will have wrong numbers in memory.

Reply:
I actually use the decremented list to carry out my if conditions, what code would you suggest to make this function optimal.